File: argentum-ui/src/app/common/rest-service/response/statistics-response.ts
import { Statistics } from '../../model/statistics';
import { QuantitySalesResponse } from './quantity-sales-response';
export class StatisticsResponse {
  guestsTotal: number;
  guestsCheckedIn: number;
  cardsTotal: number;
  totalPositiveBalance: number;
  totalNegativeBalance: number;
  totalBonus: number;
  totalSpent: number;
  totalRefund: number;
  totalDeposited: number;
  totalWithdrawn: number;
  numProducts: number;
  numLegacyProducts: number;
  numRanges: number;
  numCategories: number;
  quantitySales: QuantitySalesResponse[];
}

export function toStatistics(response: StatisticsResponse): Statistics {
  return {
    guestsTotal: response.guestsTotal,
    guestsCheckedIn: response.guestsCheckedIn,
    cardsTotal: response.cardsTotal,
    totalPositiveBalance: response.totalPositiveBalance,
    totalNegativeBalance: response.totalNegativeBalance,
    totalBonus: response.totalBonus,
    totalSpent: response.totalSpent,
    totalRefund: response.totalRefund,
    totalDeposited: response.totalDeposited,
    totalWithdrawn: response.totalWithdrawn,
    numProducts: response.numProducts,
    numLegacyProducts: response.numLegacyProducts,
    numRanges: response.numRanges,
    numCategories: response.numCategories,
    quantitySales: response.quantitySales.map(
      (sales: QuantitySalesResponse) => [sales.product, sales.quantity] as [number, number]
    ).sort((a: [number, number], b: [number, number]) => a[1] - b[1])
  };
}

File: argentum-api/src/main/java/net/devium/argentum/rest/model/response/StatisticsResponse.java
package net.devium.argentum.rest.model.response;

import java.math.BigDecimal;
import java.util.List;

public class StatisticsResponse {
    ;

    private final long guestsTotal;
    private final long guestsCheckedIn;
    private final long cardsTotal;
    private final BigDecimal totalPositiveBalance;
    private final BigDecimal totalNegativeBalance;
    private final BigDecimal totalBonus;
    private final BigDecimal totalSpent;
    private final BigDecimal totalRefund;
    private final BigDecimal totalDeposited;
    private final BigDecimal totalWithdrawn;
    private final long numProducts;
    private final long numLegacyProducts;
    private final long numRanges;
    private final long numCategories;
    private final List<QuantitySalesResponse> quantitySales;

    public StatisticsResponse(
            long guestsTotal,
            long guestsCheckedIn,
            long cardsTotal,
            BigDecimal totalPositiveBalance,
            BigDecimal totalNegativeBalance,
            BigDecimal totalBonus,
            BigDecimal totalSpent,
            BigDecimal totalRefund,
            BigDecimal totalDeposited,
            BigDecimal totalWithdrawn,
            long numProducts,
            long numLegacyProducts,
            long numRanges,
            long numCategories,
            List<QuantitySalesResponse> quantitySales
    ) {
        this.guestsTotal = guestsTotal;
        this.guestsCheckedIn = guestsCheckedIn;
        this.cardsTotal = cardsTotal;
        this.totalPositiveBalance = totalPositiveBalance;
        this.totalNegativeBalance = totalNegativeBalance;
        this.totalBonus = totalBonus;
        this.totalSpent = totalSpent;
        this.totalRefund = totalRefund;
        this.totalDeposited = totalDeposited;
        this.totalWithdrawn = totalWithdrawn;
        this.numProducts = numProducts;
        this.numLegacyProducts = numLegacyProducts;
        this.numRanges = numRanges;
        this.numCategories = numCategories;
        this.quantitySales = quantitySales;
    }

    public long getGuestsTotal() {
        return guestsTotal;
    }

    public long getGuestsCheckedIn() {
        return guestsCheckedIn;
    }

    public long getCardsTotal() {
        return cardsTotal;
    }

    public BigDecimal getTotalPositiveBalance() {
        return totalPositiveBalance;
    }

    public BigDecimal getTotalNegativeBalance() {
        return totalNegativeBalance;
    }

    public BigDecimal getTotalBonus() {
        return totalBonus;
    }

    public BigDecimal getTotalSpent() {
        return totalSpent;
    }

    public BigDecimal getTotalRefund() {
        return totalRefund;
    }

    public BigDecimal getTotalDeposited() {
        return totalDeposited;
    }

    public BigDecimal getTotalWithdrawn() {
        return totalWithdrawn;
    }

    public long getNumProducts() {
        return numProducts;
    }

    public long getNumLegacyProducts() {
        return numLegacyProducts;
    }

    public long getNumRanges() {
        return numRanges;
    }

    public long getNumCategories() {
        return numCategories;
    }

    public List<QuantitySalesResponse> getQuantitySales() {
        return quantitySales;
    }
}
